views/quiz.py

```diff
@@ -0,0 +1,307 @@
+                            f"{'🕘 Avg Time:':<12} {func.convert_seconds(average_time)} {'🔺' if average_time < state['average_time'] else '🔻'}\n" \
+                            f"{'🔥 Points:':<12} {state['points']} ({'+' if total_points >= 0 else '-'}{abs(total_points)})```"
+
+        await func.update_user(self.author.id, {"$set": {"game_state.quiz_game": state}})
+        await self.response.edit(content="This quiz has expired.", embed=embed, view=None)
+        self.stop()
+        
+    def build_embed(self) -> discord.Embed:
+        question: Question = self.currect_question
+
+        embed = discord.Embed(title=f"Question {self.current + 1} out of {len(self.questions)}", color=QUIZ_LEVEL_BASE.get(question.level)[1][2])
+        embed.description = f"**Answer Time: <t:{round(time.time() + question.average_time)}:R>**\n```{question.question}```"
+        if question.attachment:
+            embed.set_image(url=question.attachment)
+
+        embed.set_footer(text=f"Correct: {question.correct_rate}% | Wrong: {question.wrong_rate}%")
+
+        self._timeout = self._answering_time + question.average_time
+        return embed
+    
+    def cal_results(self) -> tuple[str, float]:
+        summary = ""
+        total_points = 0.0
+        symbols = {True: "✅", False: "❌", None: "⬛"}
+
+        for question, result in zip(self.questions, self._results):
+            summary += symbols[result] + " "
+
+            points = QUIZ_LEVEL_BASE.get(question.level)[1]
+            total_points += points[0] if result else -points[1] if result is False else -points[1] * (1 - .5)
+
+            if result is True:
+                question._correct += 1
+            else:
+                question._wrong += 1
+
+        return summary, total_points
+
+    def gen_response(self, result: bool = None) -> str:
+        response_base = QUESTION_RESPONSE_BASE.get(result)
+        response = f"<:{choice(response_base.get('emojis'))}> {choice(response_base.get('responses'))} "
+        if (self.current + 1) < len(self.questions):
+            response += choice(QUESTION_RESPONSE_BASE.get('next_question'))
+
+        return response
+
+    @property
+    def total_time(self) -> float:
+        return sum([question.average_time for question in self.questions]) + (len(self.questions) * self._delay_between_questions)
+    
+    @property
+    def used_time(self) -> float:
+        return round(self._ended_time - self._start_time, 2)
+    
+    @property
+    def currect_question(self) -> Question:
+        return self.questions[self.current]
+
+    @discord.ui.button(label="Answer", style=discord.ButtonStyle.green)
+    async def answer(self, interaction: discord.Interaction, button: discord.ui.Button) -> None:
+        if self._results[self.current] is not None:
+            return await interaction.response.send_message("You are already answered! Please for the next question.", ephemeral=True, delete_after=5)
```

Comment:
typos. could change it to "You've already responded to this question! Please wait for the next one."